test/relayHub/RelayHub.test.ts

```diff
@@ -1,1177 +1,507 @@
-import { Penalizer } from 'typechain-types/contracts/Penalizer';
-import { RelayHub } from 'typechain-types/contracts/RelayHub';
-import { createContractDeployer } from './utils';
-import { ERR_HUB_BAD_PARAMS, ERR_NOT_OWNER, ERR_TOO_MANY_WORKERS, ERR_UNSTAKED, ERR_WORKER_HAS_MANAGER } from '../utils/errorMessages.utils';
+import {BigNumber} from 'ethers';
+import {ethers} from 'hardhat';
+import {Penalizer} from 'typechain-types/contracts/Penalizer';
+import {RelayHub} from 'typechain-types/contracts/RelayHub';
+import {createContractDeployer} from './utils';
+
+import {
+  ERR_HUB_BAD_PARAMS,
+  ERR_NOT_OWNER,
+  ERR_TOO_MANY_WORKERS,
+  ERR_UNSTAKED,
+  ERR_WORKER_HAS_MANAGER
+} from '../utils/errorMessages.utils';
 
 chai.use(smock.matchers);
 chai.use(chaiAsPromised);
 
-describe('RelayHub Contract', function() {
-  let deployRelayHubContract: ReturnType<typeof createContractDeployer>;
-  let relayManager: SignerWithAddress, relayManagerAddr: string;
-  let relayWorker: SignerWithAddress, relayWorkerAddr: string;
-  let otherUsers: SignerWithAddress[];
+describe('RelayHub Contract', function () {
+    let deployRelayHubContract: ReturnType<typeof createContractDeployer>;
+    let relayManager: SignerWithAddress, relayManagerAddr: string;
+    let relayWorker: SignerWithAddress, relayWorkerAddr: string;
+    let relayOwner: SignerWithAddress, relayOwnerAddr: string;
+    let otherUsers: SignerWithAddress[];
 
-  let fakePenalizer: FakeContract<Penalizer>;
-  let mockRelayHub: MockContract<RelayHub>;
+    let fakePenalizer: FakeContract<Penalizer>;
+    let mockRelayHub: MockContract<RelayHub>;
 
-  beforeEach(async function() {
-    [, relayManager, relayWorker, ...otherUsers] =
-      await ethers.getSigners();
-    relayManagerAddr = relayManager.address;
-    relayWorkerAddr = relayWorker.address;
+    beforeEach(async function () {
+      [relayOwner, relayManager, relayWorker, ...otherUsers] =
+        await ethers.getSigners();
+      relayManagerAddr = relayManager.address;
+      relayWorkerAddr = relayWorker.address;
+      relayOwnerAddr = relayOwner.address;
 
-    fakePenalizer = await smock.fake('Penalizer');
+      fakePenalizer = await smock.fake('Penalizer');
 
-    deployRelayHubContract = createContractDeployer(fakePenalizer.address);
-  });
+      deployRelayHubContract = createContractDeployer(fakePenalizer.address);
 
-  afterEach(function() {
-    mockRelayHub = undefined as unknown as MockContract<RelayHub>;
-  });
+      mockRelayHub = await deployRelayHubContract();
+    });
 
-  describe('constructor', function() {
-    it('should require maxWorkerCount > 0', async function() {
-      const promiseOfRevert = deployRelayHubContract([undefined, '0']);
+    afterEach(function () {
+      mockRelayHub = undefined as unknown as MockContract<RelayHub>;
+    });
 
-      await expect(promiseOfRevert).to.have.revertedWith(
-        ERR_HUB_BAD_PARAMS
-      );
+    describe('constructor', function () {
```

Comment:
Here all the fail scenarios are being tested but not the happy path. You are testing some kind of success in every test case but only when is not rejected with ERR_HUB_BAD_PARAMS.